### ISSUE TITLE
Countdown not allowing dynamic value

### ISSUE THREAD

**Issue by analuisamartins** [siddii/angular-timer#97]
Hi,
I've not found a way to add a dynamic(scope) value in attr countdown. 
Ex:  <timer countdown="timeLimit"/> or <timer countdown="{{ timeLimit }}"/> (This way throws error in parsing)
To fix this I changed the module directive adding the last line:
 (countdown: '=', and it works) 

```
  scope: {
    interval: '=interval',
    startTimeAttr: '=startTime',
    endTimeAttr: '=endTime',
    countdownattr: '=countdown',
    countdown: '=',
```

I'm missing something? There is another way?
Thanks.

**Comment by FelixFortis** [comment]
Adding an `ng-if` worked for me, ensuring the variable was ready to go before the directive was rendered.

So from:

    <timer end-time="myVariable">{{minutes}} minute{{minutesS}}</timer>

To:

    <timer ng-if="myVariable" end-time="myVariable">{{minutes}} minute{{minutesS}}</timer>

My example hasn't been tested for `countdown`, but I'm pretty sure it will translate. 

I don't think that adding `countdown: '='` would be a good way to do things, as this is already set up with `countdownattr: '=countdown'`, and it is `countdownattr` that does all the magic. 

You just need to delay rendering the directive until the point in the `$digest` cycle when your variable has been evaluated with an `ng-if`.

Hope this helps.

**Comment by papasax** [comment]
Yes this was the reason. The variable was not set yet. I fixed it using a django template variable instead of a angular variable which was not always set. 
Your method with ng-if sounds good too.

**Comment by pencilcheck** [comment]
@FelixFortis I tried that before, but the problem with that is first off end-time doesn't work at all, second, the moment it is assigned it will start with whatever time that was last rendered, then takes about a sec to update to the current value, meaning you will get a 1 or 2 sec flash. So this is not optimal in my opinion.

**Comment by h3nr1ke** [comment]
+1 for ng-if solution....

**Comment by wtczk** [comment]
works fine:
`<timer ng-if="vm.countdown" countdown="vm.countdown"`

NaN:
`<timer countdown="vm.countdown"`

+1 for ng-if solution

**Comment by Miljokodiyan** [comment]
+1 for ng-if solution ..i tried many ways except ng-if...Wowww its worked perfectly

**Comment by hdung912** [comment]
👍  for @wtczk  it work for me thanks you :)
